.github/workflows/publish-release.yml

```diff
@@ -0,0 +1,183 @@
+          done
+      - name: Process property for matrix
+        id: processmatrix
+        run: |
+          if ! ${{ steps.properties.outputs.release-extra-branch }}; then
+            echo "::set-output name=matrix-to-exclude::Snapshots"
+          fi
+  Build-And-Publish:
+    runs-on: ubuntu-latest
+    needs: [Get-Properties]
+    strategy:
+      matrix:
+        branch: [Release, Snapshots]
+        exclude:
+          - branch: ${{ needs.Get-Properties.outputs.matrix-exclude-branch }}
+    steps:
+      - name: Get info from branch to run
+        id: getbranchinfo
+        run: |
+          if ${{ matrix.branch == 'Snapshots'}}; then
+            echo "::set-output name=branchname::${{ needs.Get-Properties.outputs.extra-branch-name }}"
+            echo "::set-output name=version::${{ needs.Get-Properties.outputs.extra-branch-curse-version }}"
+          else
+            echo "::set-output name=version::${{ needs.Get-Properties.outputs.minecraft-version }}"
+          fi
+      - name: Checkout the sources
+        uses: actions/checkout@v2
+        with:
+          ref: ${{ steps.getbranchinfo.outputs.branchname }}
+      - name: Set up JDK 1.8
+        uses: actions/setup-java@v1
+        with:
+          java-version: 1.8
+      - name: Cache Gradle packages
+        uses: actions/cache@v2
+        with:
+          path: ~/.gradle/caches
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle') }}
+          restore-keys: ${{ runner.os }}-gradle
+      - name: Grant execute permission to gradlew
+        run: chmod +x gradlew
+      - name: Build with Gradle
+        run: ./gradlew build
+      - name: Find correct JAR
+        id: findjar
+        run: |
+          output="$(find build/libs/ ! -name "*-dev.jar" ! -name "*-sources.jar" -type f -printf "%f\n")"
+          echo "::set-output name=jarname::$output"
+      - name: Save build artifacts in the action
+        uses: actions/upload-artifact@v2
+        with:
+          name: Artifacts for ${{ matrix.branch }}
+          path: build/libs
+      - name: Upload to the Github release
+        uses: actions/upload-release-asset@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          upload_url: ${{ github.event.release.upload_url }}
+          asset_path: build/libs/${{ steps.findjar.outputs.jarname }}
+          asset_name: ${{ steps.findjar.outputs.jarname }}
+          asset_content_type: application/java-archive
+      - name: Upload to Curseforge
+        uses: itsmeow/curseforge-upload@v3
+        with:
+          token: ${{ secrets.CF_API_TOKEN }}
+          project_id: 349239
+          game_endpoint: minecraft
+          file_path: build/libs/${{ steps.findjar.outputs.jarname }}
+          changelog_type: markdown
+          changelog: ${{ github.event.release.body }}
+          display_name: Carpet Mod v${{ needs.Get-Properties.outputs.carpet-version }} for ${{ steps.getbranchinfo.outputs.version }}
```

Comment:
Since 1.17 snapshots, this may be too vague (cf version for all snapshots is `1.17`).

Should I add an entry for an extra word/words after the CF version?